### PR TITLE
fix: Apply suggested fix from @fcoelho to support podman

### PR DESCRIPTION
**Pull request by rosskarchner**
The code comes from this comment: https://github.com/aws/aws-cdk/issues/16209#issuecomment-913494909

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by rosskarchner** [comment]
I don't think the build failures have anything to do with the change I've introduced

File: packages/cdk-assets/lib/private/docker.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export class Docker {
     await this.execute(['login',
       '--username', credentials.username,
       '--password-stdin',
-      credentials.endpoint], {
+      credentials.endpoint.replace(/^https?:\/\//, '')], {
```
Comment (Ragnoroct) [inline packages/cdk-assets/lib/private/docker.ts:73]:
```suggestion
      credentials.endpoint.replace(/^https?:\/\/|\/$/g, '')], {
```

Remove end slash too considering that the test data included an end slash in the endpoint.

**Review by Ragnoroct — CHANGES_REQUESTED** [review]

File: packages/cdk-assets/lib/private/docker.ts (hunk 1 above)
Comment (Ragnoroct) [reply packages/cdk-assets/lib/private/docker.ts:73]: @rosskarchner Tests also need to be fixed in `packages/cdk-assets/test/docker-images.test.ts`

Here is one test that is failing
https://github.com/aws/aws-cdk/blob/ac9901ada20e0bcadcae0e6f59e5c58220328714/packages/cdk-assets/test/docker-images.test.ts#L154

I believe there are four total that use `https`

Comment (rosskarchner) [reply packages/cdk-assets/lib/private/docker.ts:73]: Thanks! I'll take a look at these

Comment (rosskarchner) [reply packages/cdk-assets/lib/private/docker.ts:73]: @Ragnoroct thanks, I'll take a look at these (once I get a working environment to test with)

**Comment by rosskarchner** [comment]
OK-- I've moved on from needing this (I'm no longer attempting to run CDK with podman), and I'm feeling a little out of my depth trying to understand the test suite. I'm going to go ahead and close this.

**Pull request closed without merging**